frontend/src/lib/components/DateFilter/DateFilterExperiment.tsx

```diff
@@ -0,0 +1,204 @@
+    dateOptions?: Record<string, dateMappingOption>
+    isDateFormatted?: boolean
+    selectProps?: SelectProps<any>
+}
+
+interface RawDateFilterProps extends DateFilterProps {
+    dateFrom?: string | null | dayjs.Dayjs
+    dateTo?: string | null | dayjs.Dayjs
+}
+
+export function DateFilterExperiment({
+    bordered,
+    defaultValue,
+    showCustom,
+    showRollingRangePicker = true,
+    style,
+    disabled,
+    makeLabel,
+    onChange,
+    getPopupContainer,
+    dateFrom,
+    dateTo,
+    dateOptions = dateMappingExperiment,
+    isDateFormatted = true,
+    selectProps = {},
+}: RawDateFilterProps): JSX.Element {
+    const logicProps = { dateFrom, dateTo, onChange, defaultValue, dateOptions, isDateFormatted }
+    const { open, openDateRange, close, setRangeDateFrom, setRangeDateTo, setDate } = useActions(
+        dateFilterLogic(logicProps)
+    )
+    const { isOpen, isDateRangeOpen, rangeDateFrom, rangeDateTo, currentKey } = useValues(dateFilterLogic(logicProps))
+
+    const optionsRef = useRef<HTMLDivElement | null>(null)
+    const rollingDateRangeRef = useRef<HTMLDivElement | null>(null)
+
+    function _onChange(v: string): void {
+        if (v === 'Date range') {
+            openDateRange()
+            if (isOpen) {
+            }
+        } else {
+            setDate(dateOptions[v].values[0], dateOptions[v].values[1])
+            close()
+        }
+    }
+
+    function onClick(): void {
+        if (isDateRangeOpen) {
+            return
+        }
+        open()
+    }
+
+    function dropdownOnClick(e: React.MouseEvent): void {
+        e.preventDefault()
+        open()
+        document.getElementById('daterange_selector')?.focus()
+    }
+
+    function onApplyClick(): void {
+        close()
+        const formattedRangeDateFrom = dayjs(rangeDateFrom).format('YYYY-MM-DD')
+        const formattedRangeDateTo = dayjs(rangeDateTo).format('YYYY-MM-DD')
+        setDate(formattedRangeDateFrom, formattedRangeDateTo)
+    }
+
+    const onClickOutside = (event: MouseEvent): void => {
+        const target = (event.composedPath?.()?.[0] || event.target) as HTMLElement
+
+        if (!target) {
+            return
+        }
+
+        const clickInOptionsDropdown = optionsRef.current?.contains(target)
+
+        const clickInDateDropdown = event
+            .composedPath?.()
+            ?.find((e) => (e as HTMLElement)?.matches?.('.rolling-date-range-options-selector-popup'))
+
+        const isNotDateOptionsSelector =
+            !showRollingRangePicker || (isOpen && showRollingRangePicker && !clickInDateDropdown)
+
+        if (isOpen && !clickInOptionsDropdown && isNotDateOptionsSelector) {
+            close()
+            return
+        }
+    }
+
+    useEffect(() => {
+        window.addEventListener('mousedown', onClickOutside)
+        return () => {
+            window.removeEventListener('mousedown', onClickOutside)
+        }
+    }, [isOpen])
```

Comment:
Made it so that `Popup` can now handle additional refs passed as props, so that it can manage nested popups, and removed all the click handling logic that was outside of it!